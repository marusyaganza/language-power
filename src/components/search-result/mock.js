export const data = {
  uuid: '0',
  name: 'demure',
  defs: [
    '{bc} quiet and polite',
    '{bc} not attracting or demanding a lot of attention {bc} not showy or flashy'
  ],
  particle: 'adjective',
  audio: 'demure01',
  pronunciation: [
    {
      transcription: 'dɪˈmjuɚ',
      audioUrl: 'https://media.merriam-webster.com/soundc11/d/demure01.wav'
    }
  ],
  stems: ['demure', 'demurely', 'demureness'],
  fullDef: {
    definition:
      '{bc}not attracting or demanding a lot of attention {bc}not showy or flashy ',
    examples: [
      'a {it}demure{/it} young lady',
      'She was wearing a {it}demure{/it} gray suit.',
      'the {it}demure{/it} charm of the cottage',
      'a {it}demure{/it} personality'
    ]
  }
};
