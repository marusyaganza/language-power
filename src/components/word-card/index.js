import {WordCardComponent} from "./word-card";
import {fetchWord} from "../../actions";
import {connect} from "react-redux";

const loadData = (store) => store.dispatch(fetchWord('noun'));

const mapStateToProps = ({words}) => ({words});

const component = connect(mapStateToProps, {fetchWord})(WordCardComponent);

export const WordCard = {component, loadData};