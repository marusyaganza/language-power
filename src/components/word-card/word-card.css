@import '../../assets/styles/colors.css';
@import '../../assets/styles/fonts.css';

.word-card {
    padding: 10px;
    border: 1px solid var(--purple-lighter);
    margin: 10px 20px;
    box-shadow: 0 3px 0 var(--gray-lightest);
    font-family: var(--font-gayathry);
}
h3 {
    margin-top: 0;
}
