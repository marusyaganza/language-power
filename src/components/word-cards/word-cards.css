.cards-list {
    display: flex;
    flex-direction: column-reverse;
}
.modal {
    margin: 20px auto;
    text-align: center;
}
