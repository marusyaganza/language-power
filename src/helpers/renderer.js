import {renderToString} from "react-dom/server";
import {StaticRouter} from 'react-router-dom';
import React from "react";
import {Routes} from "../Routes";

export function renderer(url) {
    const content = renderToString(
        <StaticRouter location={url} context={{}}>
            <Routes/>
        </StaticRouter>
    );
    return `
    <html lang="en">
        <head>
            <title>Language power</title>
            <link rel="stylesheet" href="main.css">
        </head>
        <body>
            <div id="root">${content}</div>
            <script src="client.js"></script>
        </body>
    </html>
    `;
}
