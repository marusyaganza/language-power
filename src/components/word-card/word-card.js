import React, { memo } from 'react';
import PropTypes from 'prop-types';
import uuid from 'uuid';
import styles from './word-card.css';
import { AudioButton } from '../audio-button/audio-button';
import { Icon } from '../icons/icon';
import { DictionaryEntity } from '../dictionary-entity/dictionary-entity';
import { ShowMore } from '../show-more/show-more';
import '../word-cards/word-cards.css';

// TODO handle empty list elem
export const WordCard = memo(({ word }) => {
  const { name, defs, particle, pronunciation, fullDef, stems } = word;
  const { examples } = fullDef;

  const renderExamples = ex => {
    if (!ex || !ex.length) return null;
    const listItems = ex.map(item => (
      <li key={uuid()} className={styles.examplesItem}>
        <DictionaryEntity className={styles.examplesListInfo} text={item} />
      </li>
    ));
    return (
      <section className={styles.examples}>
        <details>
          <summary className={styles.examplesTitle}>Examples</summary>
          <ol className={styles.examplesList}>{listItems}</ol>
        </details>
      </section>
    );
  };
  const renderAudio = data => {
    if (!data.length) return null;
    const { transcription, audioUrl } = data[0];
    return audioUrl ? (
      <AudioButton buttonText={`[${transcription}]`} src={audioUrl} />
    ) : (
      <section className={styles.audio}>
        <div>[{transcription}]</div>
      </section>
    );
  };
  const renderDefs = defsArr => {
    if (!defsArr) return null;
    return defsArr.map(def => {
      return defsArr.length ? (
        // TODO create list component for this
        <li className={styles.defListItem} key={uuid()}>
          <i className={styles.textIcon}>
            <Icon width={14} height={16} id="book" />
          </i>
          <p className={styles.defItem}>
            <DictionaryEntity text={def} />
          </p>
        </li>
      ) : null;
    });
  };

  return (
    <article className={styles.word}>
      <header className={styles.wordHeader}>
        <span className={styles.name}>{name}</span>
        <span className={styles.particle}>{particle}</span>{' '}
        {renderAudio(pronunciation)}
      </header>
      <ul className={styles.defList}>{renderDefs(defs)}</ul>
      <ShowMore items={stems} title="word forms: " initialNumber={5} />
      {renderExamples(examples)}
    </article>
  );
});

WordCard.propTypes = {
  word: PropTypes.object.isRequired
};
