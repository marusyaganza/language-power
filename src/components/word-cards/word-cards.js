import React, { useContext } from 'react';
import './word-cards.css';
import { WordCard } from '../word-card/word-card';
import { AppContext } from '../../app-context/appContext';

export const WordCards = () => {
  const { wordCards, deleteWord } = useContext(AppContext);
  return (
    <section className="word-cards">
      <h2 className="words-heading">
        Words: <em className="word-count">{wordCards.length}</em>
      </h2>
      {wordCards.map(word => (
        <WordCard key={word.uuid} word={word} deleteWord={deleteWord} />
      ))}
    </section>
  );
};
