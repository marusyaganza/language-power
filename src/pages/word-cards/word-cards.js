import React from 'react';
import { Words } from '../../widgets/words/words';
import { wordCardsMock } from '../../../mocks/word-cards.mock';

export const WordCardsPage = () => {
  return (
    <main>
      <h1 className="main-heading">Word cards</h1>
      <Words initialState={wordCardsMock} />
    </main>
  );
};
