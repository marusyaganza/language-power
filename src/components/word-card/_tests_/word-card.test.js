import renderer from 'react-test-renderer';
import React from 'react';
import { WordCard } from '../word-card';
import { wordCardsMock } from '../../../../mocks/word-cards.mock';

describe('SearchResult', () => {
  it('should match snapshot', () => {
    const component = renderer
      .create(<WordCard word={wordCardsMock[0]} />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
