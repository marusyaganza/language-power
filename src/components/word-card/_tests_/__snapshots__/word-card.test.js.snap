// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResult should match snapshot 1`] = `
<article>
  <header>
    <span>
      demure
    </span>
    <span>
      adjective
    </span>
     
    <div>
      <button
        onClick={[Function]}
        type="button"
      >
        <span>
          [dɪˈmjuɚ]
        </span>
        <svg
          className={null}
          height={18}
          width={18}
        >
          <use
            xlinkHref="function SvgSprite(props) {
  return React.createElement(
    'svg', 
    Object.assign({}, props, {'data-file-name': SvgSprite.name})
  );
}#play"
          />
        </svg>
      </button>
    </div>
  </header>
  <ul>
    <li>
      <i>
        <svg
          className={null}
          height={16}
          width={14}
        >
          <use
            xlinkHref="function SvgSprite(props) {
  return React.createElement(
    'svg', 
    Object.assign({}, props, {'data-file-name': SvgSprite.name})
  );
}#book"
          />
        </svg>
      </i>
      <p>
         quiet and polite 
      </p>
    </li>
    <li>
      <i>
        <svg
          className={null}
          height={16}
          width={14}
        >
          <use
            xlinkHref="function SvgSprite(props) {
  return React.createElement(
    'svg', 
    Object.assign({}, props, {'data-file-name': SvgSprite.name})
  );
}#book"
          />
        </svg>
      </i>
      <p>
         not attracting or demanding a lot of attention  not showy or flashy 
      </p>
    </li>
  </ul>
  <section
    className="show-more"
  >
    <p>
      <h3>
        word forms: 
      </h3>
      demure, demurely, demureness
    </p>
  </section>
  <section>
    <details>
      <summary>
        Examples
      </summary>
      <ol>
        <li>
          a 
          <i>
             
            demure 
             
          </i>
          young lady 
        </li>
        <li>
          She was wearing a 
          <i>
             
            demure 
             
          </i>
          gray suit. 
        </li>
        <li>
          the 
          <i>
             
            demure 
             
          </i>
          charm of the cottage 
        </li>
        <li>
          a 
          <i>
             
            demure 
             
          </i>
          personality 
        </li>
      </ol>
    </details>
  </section>
</article>
`;
